apps/server/src/modules/tool/external-tool/uc/external-tool.uc.ts

```diff
@@ -71,4 +96,29 @@ export class ExternalToolUc {
 		const user: User = await this.authorizationService.getUserWithPermissions(userId);
 		this.authorizationService.checkAllPermissions(user, [permission]);
 	}
+
+	async getExternalToolBinaryLogo(toolId: EntityId): Promise<ExternalToolLogo> {
+		const tool: ExternalTool = await this.externalToolService.findExternalToolById(toolId);
+
+		if (!tool.logoBase64) {
+			throw new ExternalToolLogoNotFoundLoggableException(toolId);
+		}
+
+		const logoBinaryData: Buffer = Buffer.from(tool.logoBase64, 'base64');
+
+		const externalToolLogoDto: ExternalToolLogo = new ExternalToolLogo({
+			contentType: this.detectContentType(logoBinaryData),
+			logo: logoBinaryData,
+		});
+
+		return externalToolLogoDto;
+	}
+
+	private detectContentType(imageBuffer: Buffer): string {
```

Comment:
This magic bytes check can be also placed into the validation imo. though we should talk about it with PO first to know all allowed file types